### DISCUSSION TITLE
How to change Ctrl+R to the up arrow key(↑) to view the command history?

### DISCUSSION THREAD

**Discussion by wilikeit** [junegunn/fzf#4130] [Q&A]
How can I change Ctrl+R to the up arrow key to view the command history after running the command "eval "$(fzf --bash)"? I want to use the up arrow key "↑" to view the command history. How can I modify the bind in ~/.bashrc?

**Comment by LangLangBart** [comment]
> How can I change Ctrl+R to the up arrow key


Ref: #3800

**Reply by wilikeit** [reply]
> > 如何将 Ctrl+R 更改为向上箭头键
> 
> 编号： #3800

Thank you！I have a question I would like to ask you.
I want to implement select a command to execute, but there is no output date on the screen, is there anything wrong with my configuration？

![image](https://github.com/user-attachments/assets/03a1adda-b00e-4cf8-9445-1f33a9adba87)

my Config
export FZF_CTRL_R_OPTS="
  --bind 'enter:execute(echo  {2..}|bash)'
  --color header:italic
  --header 'Press  enter to execute cmd"

**Reply by LangLangBart** [reply]
> but there is no output date on the screen

When you use the `height` flag, which `fzf` sets in `__fzf_defaults` to 40%, the output from your executed command won't appear once you exit `fzf`. 

**EDIT1:** Include permalink to `__fzf_defaults`

https://github.com/junegunn/fzf/blob/ec3acb19321b8f74035225baf4a188cbb1163909/shell/key-bindings.bash#L20-L26

Try the commands below, notice the first command wont show the output of `date`.

**EDIT2:** Ref: https://github.com/junegunn/fzf/issues/2015

```bash
# nothing gets printed to the terminal with height less than 100%
fzf --height 40% --bind 'start:execute(date)+accept'

# date is printed; last flag overwrites previous flag
fzf --height 40% --bind 'start:execute(date)+accept' --height 100%
Fri Dec 13 10:28:18 CET 2024

fzf --height 40% --bind 'start:execute(date)+accept' --no-height
Fri Dec 13 10:28:28 CET 2024
```

```diff
--- a
+++ b
export FZF_CTRL_R_OPTS="
 --bind 'enter:execute(echo {2..}|bash)'
 --color header:italic
 --header 'Press enter to execute cmd'
+--no-height"
```

---

> I want to implement select a command to execute

I assume you want to 'execute' the command from your shell history directly.

Seems to be not possible for `bash` users. 

Ref: https://github.com/junegunn/fzf/issues/609#issuecomment-230339037

https://github.com/junegunn/fzf/blob/ec3acb19321b8f74035225baf4a188cbb1163909/shell/key-bindings.bash#L56-L105

**Reply by wilikeit** [reply]
Thanks @LangLangBart , I successfully executed the historical command, but "  --no-height"  was really hard to notice.


```bash
export FZF_CTRL_R_OPTS="
  --bind 'enter:execute(echo -n {2..}|bash)+abort'  
  --color header:italic
  --header 'Press  enter to execute cmd'
  --no-height"
```

**Reply by LangLangBart** [reply]
Good. Just be aware of the limitations when using `FZF_CTRL_R_OPTS` in this manner, as the executed command may behave differently than expected.

>[!NOTE] 
> _fzf starts child processes using `$SHELL -c` if `$SHELL` is set, otherwise sh -c._


This causes your shell to fork, losing access to the original shell environment, including aliases. For example:

```bash
alias date='date +%d/%b/%y'
date
# 13/Dec/24

fzf --bind 'start:execute(date)+accept'
# Fri Dec 13 13:44:16 CET 2024
```

**Reply by wilikeit** [reply]
Thanks for the reminder. I set it up below
export SHELL=/bin/bash
